### PR TITLE
Enable licence usage data for Infra Server.

### PR DESCRIPTION
**Pull request by sreepuramsudheer**
### Description

As part of this PR telemetry functionality introduced to chef-server, telemetry service runs once in a day collects below information.
* Version of Infra Server
* FQDN of the load balancer Chef Server is installed on
* Email domain of the users registered to Chef Server
* 30 days average usage on the number of nodes check in on daily basis
* Binary file name and location of Chef Infra Server command
* Configuration file location of Chef Infra Server

### Issues Resolved

Related tickets:
https://chefio.atlassian.net/browse/CHEF-12623

Documentation:
https://chefio.atlassian.net/browse/CHEF-14681
https://github.com/chef/chef-server/pull/3870 

### Check List

- [x] New functionality includes tests
- [x] All buildkite tests pass
- [x] Full omnibus build and tests in buildkite pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by sreepuramsudheer** [comment]
Adhoc: https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/6469

File: src/oc_erchef/apps/chef_telemetry/rebar.config [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+{deps,
+    [
+    %% lager has to come first since we use its parse transform
+    {lager, ".*",
+        {git, "https://github.com/erlang-lager/lager",                {branch, "master"}}},
+    {opscoderl_httpc, ".*",
+        {git, "https://github.com/chef/opscoderl_httpc",              {branch, "main"}}},
+    {pooler, ".*",
+        {git, "https://github.com/chef/pooler",                       {branch, "master"}}}
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/rebar.config:13]: If we are not using pooler, we can remove it

File: src/oc_erchef/apps/chef_telemetry/src/chef_telemetry.erl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+-module(chef_telemetry).
+-export([
+         ping/0,
+         is_enabled/0,
+         token/0
+        ]).
+
+-spec ping() -> pong | pang.
+ping() ->
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/src/chef_telemetry.erl:30]: Were we able to see the chef-telemetry app status in our chef-server status endpoint ?
Should we disable the chef-telementry in standalone automate ?

File: src/oc_erchef/apps/chef_telemetry/src/chef_telemetry.erl [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    case application:get_env(chef_telemetry, root_url) of
+        {ok, _Value} ->
+            true;
+        undefined ->
+            false
+    end.
+
+-spec token() -> string() | atom().
+token() ->
+    case chef_secrets:get(<<"data_collector">>, <<"token">>) of
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/src/chef_telemetry.erl:47]: Not sure what we are doing with data_collector token ?

File: src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+    State#state{
+        current_scan = CurrentScan#current_scan{
+            total_nodes = Count,
+            active_nodes = ActiveNodes}}.
+
+generate_request(ServerVersion, State) ->
+    CurrentScan = State#state.current_scan,
+    jiffy:encode({[
+    {<<"licenseId">>, <<"Infra-Server-license-Id">>},
+    %%{<<"customerId">>, <<"">>},
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl:271]: delete

File: src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+                    sqerl:adhoc_insert(telemetry, [[{<<"property">>, <<"last_send">>}, {<<"valuestring">>, Node}, {<<"timestamp1">>, Now}]]),
+                    true;
+                false ->
+                    false
+            end;
+        {ok, Rows} when is_list(Rows) andalso length(Rows) == 0 ->
+            sqerl:adhoc_insert(telemetry, [[{<<"property">>, <<"last_send">>}, {<<"valuestring">>, Node}, {<<"timestamp1">>, Now}]]),
+            true;
+        Error -> 
+            throw({not_able_to_gead_from_db, Error})
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl:352]:
```suggestion
            throw({not_able_to_read_from_db, Error})
```

File: src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+            case should_send(LastSend, State) of
+                true ->
+                    sqerl:adhoc_delete(telemetry, {<<"property">>, equals, <<"last_send">>}),
+                    sqerl:adhoc_insert(telemetry, [[{<<"property">>, <<"last_send">>}, {<<"valuestring">>, Node}, {<<"timestamp1">>, Now}]]),
+                    true;
+                false ->
+                    false
+            end;
+        {ok, Rows} when is_list(Rows) andalso length(Rows) == 0 ->
+            sqerl:adhoc_insert(telemetry, [[{<<"property">>, <<"last_send">>}, {<<"valuestring">>, Node}, {<<"timestamp1">>, Now}]]),
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl:349]: what is `timestamp1`

File: src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+        end,
+    lists:foldl( MaxFun, 0, SystemTimes).
+
+should_send(LastSend, State) ->
+    LastSend < State#state.scan_time.
+
+insert_fqdn(State) ->
+    {ok, HostName} = inet:gethostname(),
+    Now = calendar:system_time_to_universal_time(State#state.scan_time, second),
+    %%Hostname = os:cmd('hostname -f'),
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/src/chef_telemetry_worker.erl:375]: --

**Comment by jashaik** [comment]
@sreepuramsudheer  Looking for sql changes for postgres table changes. Any other PR to be reviewed for these sql changes?

**Comment by jashaik** [comment]
@sreepuramsudheer Needs to add tests.

**Comment by jashaik** [comment]
Automate pipeline : Green
https://buildkite.com/chef/chef-automate-main-verify-private/builds/35288

File: src/oc_erchef/apps/chef_telemetry/test/chef_telemetry_worker_test.erl [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+-export([
+    get_execute/1,
+    adhoc_select/3,
+    count_nodes/1,
+    chef_db_list/2,
+    org_metadata/2,
+    index_search/1
+]).
+%%-record(oc_chef_organization, {
```
Comment (jashaik) [inline src/oc_erchef/apps/chef_telemetry/test/chef_telemetry_worker_test.erl:13]: remove the commented changes

**Review by jashaik — APPROVED** [review]

**Pull request merged**